### ISSUE TITLE
Routing not working on a certain domain

### ISSUE THREAD

**Issue by sergeyfilimonov** [gridsome/gridsome#364]
Routing and switching between pages work fine here https://hrnetologyru.sfilimonov.now.sh/ but stop working here https://hr.netology.ru/ This is all ONE deployment. So the issue is there only with a certain domain.

What I've done to debug:
1. Checked all .vue files for markup errors.
2. Checked HTML with validator and solved some errors and warnings.
3. Added `chainWebpack: config => config.mode('development')` into https://github.com/sergeyfilimonov/hr.netology.ru/blob/master/gridsome.config.js

Code is here https://github.com/sergeyfilimonov/hr.netology.ru

**Comment by hjvedvik** [comment]
Looks like something is minifying the HTML on https://hr.netology.ru. HTML comments are missing on that site which is present on https://hrnetologyru.sfilimonov.now.sh. Vue depends on those comments to hydrate the markup correctly.

**Comment by sergeyfilimonov** [comment]
@hjvedvik I have tried to upload the same files to an FTP with the same domain hr.netology.ru and it doesn't work. It seems it means that the problem is with the code itself, it's somehow doesn't like the domain :/ Or...

Do some settings of the domain might cause this issue?

**Comment by sergeyfilimonov** [comment]
@hjvedvik we have fixed this issue. I don't know what we did exactly to fix it but it root of the problem was in some proxy/HTTPS settings or — more likely — with the fact that the main domain netology.ru is configured to work with CloudFare CDN, and these settings were applied for hr.netology.ru also. It seems Amazon and CloudFare CDNs were interfering somehow and this caused troubles with navigation.
